Choosing models - copyedits (#205)
mlsa-book/MLSA: book/P3C18_choosing_models.qmd

In contrast to measure selection, selecting models is more straightforward and the same heuristics from regression and classification largely apply to survival analysis.
First, for low-dimensional data\index{low-dimensional data}, many experiments have demonstrated that machine learning may not improve upon more standard statistical methods [@Christodoulou2019] and the same holds for survival analysis [@Burk2026].
Therefore, the cost that comes with using machine learning --- lower interpretability and longer training times --- is unlikely to provide any performance benefits when a dataset has relatively few covariates especially when combined with low sample size.
In settings where machine learning is more useful, the choice largely falls into the four model classes discussed in this book: random survival forests, survival support vector machines, gradient boosting machines, and neural networks (deep learning).

> In benchmark experiments with sufficient data and computational resources, the common strategy of including models of different complexity (featureless, linear, non-linear and interactions) is also sensible in the survival context.

Author: ```suggestion
In benchmark experiments with sufficient data and computational resources, it is sensible to include models of varying complexity, from featureless and linear models to models that capture non-linear effects and interactions.
```

Random survival forests and boosting methods are strong all-purpose methods that can handle different censoring types and competing risks settings.\index{random survival forests}\index{gradient boosting machines}
In single-event settings, both have been shown to perform well on high-dimensional data, outperforming other model classes [@Spooner2020].\index{high-dimensional data}
Forests often work well without hyperparameter tuning and may therefore be a sensible first choice for high-dimensional data.

Survival support vector machines have generally shown limited empirical success and appear to have seen little real-world adoption; moreover, their runtime can be substantial, taking hours to produce models that may see little (if any) benefit over foundational alternatives [@Fouodo2018; @Polsterl2015; @Burk2026].
Consequently, they are not generally recommended as a first-choice method.\index{survival support vector machines}

Among machine learning methods, deep learning has become the dominant paradigm across many domains, and current research trends point to a similar trajectory in survival analysis [@Wiegrebe2024].\index{neural networks}\index{deep learning}

> Neural-network performance nonetheless remains highly data-dependent: there are clear situations in which they are preferred or required --- for example when handling image data such as MRI scans, or when identifying patterns in very large datasets such as omics data --- yet there are still no firm heuristics for selecting one architecture over another.

Author: ```suggestion
Neural network performance remains highly data-dependent.
There are clear situations in which they are preferred or required, for example when handling image data such as MRI scans, or when identifying patterns in very large datasets such as omics data, yet there are no firm heuristics for selecting one architecture over another.
```

While deep learning has traditionally required large amounts of data, this constraint is increasingly relaxed by *transfer learning*, in which large *pre-trained* models are *fine-tuned* to a specific context, making powerful architectures usable on smaller datasets, particularly for feature extraction.\index{transfer learning}
This is especially valuable in multi-modal settings, where image or text inputs are combined with tabular covariates.\index{multi-modal data}
The development of *foundation models* opens further avenues, potentially leveraging general representations to benefit low-resource time-to-event problems.\index{foundation model}

> A recent example are *prior-data fitted networks* (PFNs), which amortize Bayesian inference into a single forward pass and have been adapted to right-censored time-to-event data, enabling individualized survival prediction without dataset-specific training or tuning [@Qi2026; @Seletkov2026].\index{prior-data fitted network}

Author: ```suggestion
A recent example is prior-data fitted networks (PFNs), which learn to approximate Bayesian inference so that predictions can be obtained in a single forward pass.
PFNs have been adapted to right-censored time-to-event data, enabling individualized survival prediction without dataset-specific training or tuning [@Qi2026; @Seletkov2026].\index{prior-data fitted network}
```

> In practice, many real-world applications still go beyond one-off analyses and require custom architectures developed with frameworks such as PyTorch [@pkgtorch] or TensorFlow [@pkgtensorflow] rather than off-the-shelf implementations.

Author: ```suggestion
In practice, many real-world applications go beyond one-off analyses and require custom architectures developed with frameworks such as PyTorch [@pkgtorch] or TensorFlow [@pkgtensorflow] rather than off-the-shelf implementations.
```
